modules/@ergo/core/components/Form/JSONSchemaForm/JSONSchemaFormTableRowWidget.vue

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+<template>
+    <div
+        class="row-widget"
+        :style="gridTemplateTemplate">
+        <Component
+            v-for="element in rowComponents"
+            :is="element.component"
+            v-bind="element.props"
+            :key="element.key"
+            @input="onValueChange" />
+        <Fab
+            :theme="secondaryTheme"
+            @click.native="onRemoveRow">
+            <template #icon="{ color }">
+                <IconDelete :fill-color="color" />
+            </template>
+        </Fab>
+    </div>
+</template>
+
+<script>
+import { toCapitalize } from '@Core/models/stringWrapper';
+import { THEME } from '@Core/defaults/theme';
+import { ARROW } from '@Core/defaults/icons';
+import IconDelete from '@Core/components/Icons/Actions/IconDelete';
+import Fab from '@Core/components/Buttons/Fab';
+
+export default {
+    name: 'JSONSchemaFormTableRowWidget',
+    components: {
+        Fab,
+        IconDelete,
+    },
+    props: {
+        properties: {
+            type: Object,
+            required: true,
+        },
+        index: {
+            type: Number,
+            required: true,
+        },
+        required: {
+            type: Array,
+            default: () => [],
+        },
+        value: {
+            type: Object,
+            default: () => ({}),
+        },
+    },
+    data() {
+        return {
+            localValue: {},
+        };
+    },
+    computed: {
+        secondaryTheme() {
+            return THEME.SECONDARY;
+        },
+        gridTemplateTemplate() {
+            return {
+                gridTemplateColumns: this.rowComponents.map((component, index) => (index % 2 === 0 ? '1fr' : 'max-content')).join(' '),
```

Comment:
may be moved this logic to variable, it is too complicated

Reply:
The variable would need to be called **gridTemplateColumns** - it's already a variable

Reply:
i know, but I thinking about move logic out of return